core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/ImmutablesPropertiesFactory.java

```diff
@@ -0,0 +1,266 @@
+                        MethodHandles.lookup().unreflect(m),
+                        findBuilderSetter(builderClass, name, propertyType));
+            } catch (IllegalAccessException | NoSuchMethodException e) {
+                throw new IllegalArgumentException("Failed to inspect method " + m, e);
+            }
+        }
+
+        private MethodHandle findBuilderSetter(final Class<?> builderClass, String name, Type type)
+        throws IllegalAccessException, NoSuchMethodException {
+            final NoSuchMethodException failure;
+            try {
+                return MethodHandles.lookup().unreflect(builderClass.getMethod(name, GenericTypes.getErasedType(type)));
+            } catch (NoSuchMethodException e) {
+                failure = e;
+            }
+            for (Method m : builderClass.getMethods()) {
+                if (m.getName().equals(name) && m.getParameterCount() == 1) {
+                    return MethodHandles.lookup().unreflect(m);
+                }
+            }
+            throw failure;
+        }
+
+        @Override
+        public PojoBuilder<T> create() {
+            final Object b = builder.get();
+            return new PojoBuilder<T>() {
+                @Override
+                public void set(String property, Object value) {
+                    Unchecked.biFunction(properties.get(property).setter::invoke).apply(b, value);
+                }
+
+                @SuppressWarnings("unchecked")
+                @Override
+                public T build() {
+                    return (T) Unchecked.function(builderBuild::invoke).apply(b);
+                }
+            };
+        }
+    }
+
+    static class ModifiablePojoProperties<T, M extends T> extends BasePojoProperties<T, M> {
+        ModifiablePojoProperties(Type type, Class<?> defn, Supplier<M> constructor) {
+            super(type, defn, constructor);
+        }
+
+        @Override
+        protected ImmutablesPojoProperty<T> createProperty(String name, Method m) {
+            final Class<M> impl = implClass();
+            final Type propertyType = GenericTypes.resolveType(m.getGenericReturnType(), getType());
+            try {
+                return new ImmutablesPojoProperty<T>(
+                        name,
+                        QualifiedType.of(propertyType, Qualifiers.getQualifiers(m)),
+                        m,
+                        isSetMethod(impl, name),
+                        MethodHandles.lookup().unreflect(m),
+                        MethodHandles.lookup().findVirtual(impl, setterName(name), MethodType.methodType(impl, GenericTypes.getErasedType(propertyType))));
+            } catch (IllegalAccessException | NoSuchMethodException e) {
+                throw new IllegalArgumentException("Failed to inspect method " + m, e);
+            }
+        }
+
+        @SuppressWarnings("unchecked")
+        private Class<M> implClass() {
+            return (Class<M>) builder.get().getClass();
```

Comment:
This will create a garbage builder instance on every invocation. But since it's only invoked once per method and cached, that's probably ok.